### PR TITLE
Fix #1788: Clarify the value returned by the AudioParam value getter.

### LINKED ISSUE #1788: Unclear what value is returned by AudioParam.value
Should `AudioParam.value` return the computed value for the parameter at `AudioContext.currentTime`? The spec is not clear on this:
https://webaudio.github.io/web-audio-api/#AudioParam-attributes

I know this issue was under discussion before (e.g. here: https://github.com/WebAudio/web-audio-api/issues/318), but I'm not sure which definition was approved.

Thanks in advance.

### PR DESCRIPTION
**Pull request by rtoy**
Define the intrinsic value and update the algorithm a little bit for
that.

Then update the rendering algorith so that the [[current value]] is
the intrinsic value, not the computed value (that consists of the
automation and any inputs to the param).

I think this could be simplified a bit more, but I didn't really do
that.  Perhaps all references to intrinsic value could be replaced by
[[current value]]?

***
<a href="https://pr-preview.s3.amazonaws.com/rtoy/web-audio-api/pull/1838.html" title="Last updated on May 13, 2019, 10:31 PM UTC (305cae1)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WebAudio/web-audio-api/1838/f47c7f5...rtoy:305cae1.html" title="Last updated on May 13, 2019, 10:31 PM UTC (305cae1)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -3888,27 +3888,32 @@ parameters</dfn> are {{AudioParam}}s that are used with other
 {{AudioParam}}s to compute a value that is then used as an input
 to compute the output of an {{AudioNode}}.
 
-The <dfn>computedValue</dfn> is the final value controlling the
-audio DSP and is computed by the audio rendering thread during each
-rendering time quantum. It MUST be internally computed as follows:
+The computation of the value of an {{AudioParam}} consists of two parts:
+	- the <dfn>intrinsic</dfn> value that is computed from the {{AudioParam/value}}
+		attribute and any <a href="#dfn-automation-event">automation events</a>
```
[the remark this answers is not in the document]
Comment (rtoy) [reply index.bs]: Done.

File: index.bs [hunk 2]
```diff
@@ -3888,27 +3888,32 @@ parameters</dfn> are {{AudioParam}}s that are used with other
...
+	- the <dfn>computedValue</dfn> that is the final value controlling the
+		audio DSP and is computed by the audio rendering thread during each
+		rendering time quantum.
```
[the remark this answers is not in the document]
Comment (rtoy) [reply index.bs]: Done

File: index.bs [hunk 3]
```diff
@@ -3888,27 +3888,32 @@ parameters</dfn> are {{AudioParam}}s that are used with other
...
+
+The <a>computedValue</a>It MUST be internally computed as follows:
```
[the remark this answers is not in the document]
Comment (rtoy) [reply index.bs]: Done.  But also rewritten in a different way.

File: index.bs [hunk 4]
```diff
@@ -3888,27 +3888,32 @@ parameters</dfn> are {{AudioParam}}s that are used with other
...
 
 <div algorithm="computation-of-value">
-	1. An <em>intrinsic</em> parameter value will be calculated at
+	1. An <a><var>intrinsic</var></a> parameter value will be calculated at
```
[the remark this answers is not in the document]
Comment (rtoy) [reply index.bs]: Done, by changing the algorithm around a little.

File: index.bs [hunk 5]
```diff
@@ -3496,10 +3496,9 @@ Attributes</h4>
 		initialized to the <code>defaultValue</code>.
 
 		Getting this attribute returns the contents of the
-		{{[[current value]]}} slot, which maintains the value of
-		this parameter at the conclusion of the most recent render
-		quantum on the audio rendering thread, or the most recently
-		assigned value if no rendering has taken place.
+		{{[[current value]]}} slot. See
+intr		[[#computation-of-value]] for the algorithm for the
```
Comment (hoch) [inline index.bs]: What's `intr` in this line?

File: index.bs [hunk 6]
```diff
@@ -3860,30 +3859,43 @@ parameters</dfn> are {{AudioParam}}s that are used with other
...
+	1. <var>paramIntrinsicValue</var> will be calculated at
 		each time, which is either the value set directly to the
-		<code>value</code> attribute, or, if there are any <a href="#dfn-automation-event">automation events</a> with times before or
-		at this time, the value as calculated from these events. When read,
-		the <code>value</code> attribute always returns the
-		<em>intrinsic</em> value for the current time. If automation events
-		are removed from a given time range, then the <em>intrinsic</em>
+		{{AudioParam/value}} attribute, or, if there are any
+		<a href="#dfn-automation-event">automation events</a>
+		with times before or
```
Comment (hoch) [inline index.bs]: nit: the line wrapping could be more consistent.

File: index.bs (hunk 5 above)
Comment (rtoy) [reply index.bs]: Oops.  That should be there.

File: index.bs [hunk 7]
```diff
@@ -3888,27 +3888,32 @@ parameters</dfn> are {{AudioParam}}s that are used with other
...
+		rendering time quantum.
+
+The <a>computedValue</a>It MUST be internally computed as follows:
 
 <div algorithm="computation-of-value">
-	1. An <em>intrinsic</em> parameter value will be calculated at
+	1. An <a><var>intrinsic</var></a> parameter value will be calculated at
 		each time, which is either the value set directly to the
-		<code>value</code> attribute, or, if there are any <a href="#dfn-automation-event">automation events</a> with times before or
+		{{AudioParam/value}} attribute, or, if there are any <a href="#dfn-automation-event">automation events</a> with times before or
```
[the remark this answers is not in the document]
Comment (rtoy) [reply index.bs]: I think this is ok now.

**Pull request merged**